src/docs-assembler/Cli/RepositoryCommands.cs

```diff
@@ -117,4 +127,59 @@ public async Task<int> BuildAll(
 			return collector.Errors + collector.Warnings;
 		return collector.Errors;
 	}
+
+	/// <param name="contentSource"> The content source. "current" or "next"</param>
+	/// <param name="DryRun"> Dry run without uploading to s3</param>
+	/// <param name="ctx"></param>
+	[Command("update-link-index-all")]
+	public async Task<int> UpdateLinkIndexAll(ContentSource contentSource, bool DryRun, Cancel ctx = default)
+	{
+		var collector = new ConsoleDiagnosticsCollector(logger, githubActionsService);
+		var assembleContext = new AssembleContext("prod", collector, new FileSystem(), new FileSystem(), null, null);
+		var cloner = new RepositorySourcer(logger, assembleContext.CheckoutDirectory, new FileSystem(), collector);
+		var dict = new ConcurrentDictionary<string, Stopwatch>();
+		var repositories = new Dictionary<string, Repository>(assembleContext.Configuration.ReferenceRepositories)
+		{
+			{ NarrativeRepository.RepositoryName, assembleContext.Configuration.Narrative }
+		};
+		await Parallel.ForEachAsync(repositories,
+			new ParallelOptions
+			{
+				CancellationToken = ctx,
+				MaxDegreeOfParallelism = Environment.ProcessorCount
+			}, async (kv, c) =>
+			{
+				var name = kv.Key.Trim();
```

Comment:
`try/catch` the loop method and emit an error over the diagnostics channel. 

That way we can still build all the repositories that do validate and update those.